shell/platform/android/android_surface_software.cc

```diff
@@ -80,17 +97,31 @@ bool AndroidSurfaceSoftware::PresentBackingStore(
     return false;
   }
 
-  // Pass the sk_surface buffer to the android FlutterView.
-  JNIEnv* env = fml::jni::AttachCurrentThread();
-
-  // Buffer will be copied into a Bitmap Java-side.
-  fml::jni::ScopedJavaLocalRef<jobject> direct_buffer(
-      env,
-      env->NewDirectByteBuffer(pixmap.writable_addr(), pixmap.getSize64()));
+  ANativeWindow_Buffer native_buffer;
+  ANativeWindow_lock(native_window_->handle(), &native_buffer, nullptr);
+
+  SkColorType color_type;
+  if (getSkColorType(native_buffer.format, &color_type)) {
```

Comment:
Our pixmap is always N32 premultiplied (See `SkSurface::MakeRasterN32Premul ` in `AcquireBackingStore`). If the window format turns out to be 565, the `drawBitmapRect` step would have to perform a conversion right? I would rather we assert that the window format is 8888 and not handle the 565 case at all. Or, in the `AcquireBackingStore`, query the native window for its format and create the 565 in the first place if the user selects the 565 type for the native window.